include/vector2d.h

```diff
@@ -5,6 +5,7 @@
 #ifndef __IRR_POINT_2D_H_INCLUDED__
 #define __IRR_POINT_2D_H_INCLUDED__
 
+#include <functional>

```

Comment:
Maybe include the STL header after the local headers? We don't pay a lot of attention to header order, but it would be a good practice.